Dropdb should be forced to close any active connections and not fail

sethhorsley opened the pull request:
This pull request includes a small change to the `restore_to_development` method in `lib/parity/backup.rb`. The change ensures that the `dropdb` command is executed with the `--force` flag to enforce the removal of the development database even if there are active connections.

sethhorsley committed:
wipe_development_database now force wipes db to stop error of users connected to db

sethhorsley committed:
Merge branch 'thoughtbot:main' into main

sethhorsley committed:
Update Ruby version to 3.4.1 and enhance backup functionality to support specific backup IDs. Added logging for restoration processes and updated tests to verify new behavior.

geoffharcourt commented on lib/parity/backup.rb:
```diff
@@ -77,21 +93,34 @@ def ensure_temp_directory_exists
+      else
+        puts "Downloading latest backup from #{from}..."
+        Kernel.system(
+          "curl -o tmp/latest.backup \"$(heroku pg:backups:url --remote #{from})\"",
```
Could you use the same curl statement in both places since a nil `backup_id` would be an empty string here? This would avoid having two pieces of interpolated string arguments to maintain going forward where we could have one.

geoffharcourt commented on lib/parity/backup.rb:
```diff
@@ -25,31 +26,46 @@ def restore
 
     alias :parallelize? :parallelize
 
+
```
Trim these down, please.

geoffharcourt commented on lib/parity/environment.rb:
```diff
@@ -90,9 +93,19 @@ def parallelize?
+      if backup_id_index && backup_id_index + 1 < arguments.length
+        arguments[backup_id_index + 1]
+      end
+    end
```
The predicate method should return a boolean result. Given that you're using the ID returned here I think the `?` should be dropped from the method name.

sethhorsley committed:
Refactor backup.rb: improve code style and add Rails environment loading

- Replace 'alias' with 'alias_method' for better Ruby style
- Standardize string continuation formatting across all Kernel.system calls
- Improve method chaining readability in development_db method
- Add Rails environment loading before reading database.yml to support credentials
- Use consistent double quotes for string literals
